test/tests/variables/tests

```diff
@@ -0,0 +1,8 @@
+[Tests]
+  design = 'syntax/Problem/index.md'
+  [./MFEMVariableSetupFromMOOSEVariables]
+    type = RunApp
+    input = mfem_variables_from_moose.i
+    requirement = 'Platypus shall have the ability to add an MFEMVariable from a user specification of a MOOSE variable, inferring the FESpace.'
+  []
+[]
```

Comment:
Yes - this test is checking that MFEM variable can be set up from the same syntax as a MOOSE variable, if they haven't explicitly declared the variable type to be `MFEMVariable`. In general, the latter is preferred, as that allows specification of the `FESpace` using the full range that MFEM supports. 

However users familiar with MOOSE will likely want to be able to switch to use MFEM without having to rewrite the Variables block in their input files for simple cases. Since we already have a method that infers the corresponding MFEM FESpace for nodal and elemental MOOSE variables, it's somewhat trivial to support the user specifying variables in this manner, reducing the barrier to entry to Platypus.

I've updated this test to check the variable names corresponding to `mfem::GridFunctions` present in the output, which should act as a regression test for this to confirm the `GridFunctions` have been created as expected.